src/modules/bulletproofs/rangeproof_impl.h

```diff
@@ -209,31 +209,45 @@ static int secp256k1_bulletproof_rangeproof_verify_impl(const secp256k1_ecmult_c
         size_t j;
 
         /* Commit to all input data: min value, pedersen commit, asset generator, extra_commit */
-        if (min_value != NULL && min_value[i] != NULL) {
-            secp256k1_sha256_initialize(&sha256);
-            secp256k1_sha256_write(&sha256, commit, 32);
-            for (j = 0; j < n_commits; j++) {
-                unsigned char vbuf[8];
-                vbuf[0] = min_value[i][j];
-                vbuf[1] = min_value[i][j] >> 8;
-                vbuf[2] = min_value[i][j] >> 16;
-                vbuf[3] = min_value[i][j] >> 24;
-                vbuf[4] = min_value[i][j] >> 32;
-                vbuf[5] = min_value[i][j] >> 40;
-                vbuf[6] = min_value[i][j] >> 48;
-                vbuf[7] = min_value[i][j] >> 56;
-                secp256k1_sha256_write(&sha256, vbuf, 8);
-            }
-            secp256k1_sha256_finalize(&sha256, commit);
-        }
+        secp256k1_sha256_initialize(&sha256);
+        secp256k1_sha256_write(&sha256, commit, 32);
         for (j = 0; j < n_commits; j++) {
-            secp256k1_bulletproof_update_commit(commit, &commitp[i][j], &value_gen[i]);
+            unsigned char vbuf[8];
+            uint64_t mv;
+            if (min_value != NULL && min_value[i] != NULL) {
+                mv = min_value[i][j];
+            } else {
+                mv = 0;
+            }
```

Comment:
Do we accept arbitrary min values on the chain? If yes, we should ensure that there is no malleability caused by `min_value[i] == NULL` leading to the same hash as `min_value[i][j] == 0`.

Reply:
On the chain the min value is always 0 or 1 as determined by consensus rules. (Specifically it is 0 for provably unspendable outputs, to allow mimblewimble kernels, and 1 otherwise.)